src/libs/deploy.js

```diff
@@ -178,11 +180,131 @@ export const deployAppleStore = async (signedApp, workspace = '/tmp/') => {
+    case 'ios':
+    {
+      applicationName = awFileName + '.ipa';
+      deviceType = '2';
+      modelId = 1;
+      break;
+    }
+    case 'android':
+    {
+      applicationName = awFileName + '.apk';
+      deviceType = '5';
+      modelId = 5;
+      break;
+    }
+    default:
+      throw new Error('Unsupported application type for airWatch deployment');
+  }
+
+  // The constant url for api:
+  const awHost = 'https://mobileconsole.gov.bc.ca';
+  const awUploadAPI = '/API/mam/blobs/uploadblob';
+  const awInstallAPI = '/API/mam/apps/internal/begininstall';
+
+  // Update the user account to a device-account:
+  const awUsernameF = await exec('security find-generic-password -w -s awUsername');
+  const awPasswordF = await exec('security find-generic-password -w -s awPassword');
+  const awTenantCodeF = await exec('security find-generic-password -w -s awCode');
+
+  // Extract value from stdout:
+  const awUsername = awUsernameF.stdout.trim().split('\n')[0];
+  const awPassword = awPasswordF.stdout.trim().split('\n')[0];
+  const awTenantCode = awTenantCodeF.stdout.trim().split('\n')[0];
+  
+  console.log(awUsername);
+  console.log(awPassword);
+  console.log(awTenantCode);
+
+  // Get app binary:
+  const signedAppPath = await fetchFileFromStorage(signedApp, workspace);
+  const appBinary = require('fs').readFileSync(signedAppPath);
+
+  console.log('Start to deploy to airwatch..');
+
+  // Step 1: Upload app as blob
+  const uploadOptions = {
+    headers: {
+      'Content-Type': 'application/octet-stream',
+      'aw-tenant-code': awTenantCode,
+      Accept: 'application/json',
+    },
+    auth: {
+      user: awUsername,
+      password: awPassword,
+    },
+    method: 'POST',
+    uri: url.resolve(awHost, awUploadAPI),
+    encoding: null,
+    body: appBinary,
+    qs: {
+      filename: signedApp,
+      organizationgroupid: awOrgID,
+    },
+  };
+
+  try {
+    const awUploadRes = await request(uploadOptions);
+
+    // get the blob id:
+    const blobID = JSON.parse(awUploadRes.toString()).Value;
+    console.log('The blob id is ' + blobID);
+
+    // Step 2: Install app to an Organization Group
+    const installOptions = {
+      headers: {
+        'content-type': 'application/json',
+        'aw-tenant-code': awTenantCode,
+      },
+      auth: {
+        user: awUsername,
+        password: awPassword,
+      },
+      method: 'POST',
+      uri: url.resolve(awHost, awInstallAPI),
+      body: {
+        BlobId: blobID,
+        DeviceType: deviceType,
+        ApplicationName: applicationName,
+        PushMode: 'OnDemand',
+        SupportedModels: {
+          Model: [{
+            ModelId: modelId,
+          }],
+        },
+      },
+      json: true,
+    };
+
+    await request(installOptions);
+    console.log('Finished deploying to airwatch...');
+    return signedAppPath;
+  } catch (err) {
+    const message = 'Unable to deploy to AirWatch';
+    logger.error(`${message}, err = ${err.message}`);
```

Comment:
Do you want to re-throw an error here?

Reply:
Error is thrown in the fn handleJob and handleDeploymentJob. Will just keep the logger.error for development phase and remove it when stable.